test/system/shoot_cp_migration/migrate_test.go

```diff
@@ -0,0 +1,257 @@
+		Scheme: kubernetes.ShootScheme,
+	}))
+	t.Shoot = *shoot
+	return
+}
+
+func initSeedsAndClients(ctx context.Context, t *ShootMigrationTest) error {
+	t.Config.SourceSeedName = *t.Shoot.Spec.SeedName
+
+	if seed, seedClient, err := t.GardenerFramework.GetSeed(ctx, t.Config.TargetSeedName); err != nil {
+		return err
+	} else {
+		t.TargetSeedClient = seedClient
+		t.TargetSeed = seed
+	}
+
+	if seed, seedClient, err := t.GardenerFramework.GetSeed(ctx, t.Config.SourceSeedName); err != nil {
+		return err
+	} else {
+		t.SourceSeedClient = seedClient
+		t.SourceSeed = seed
+	}
+	return nil
+}
+
+func beforeMigration(ctx context.Context, t *ShootMigrationTest, guestBookApp *applications.GuestBookTest, testSecret *corev1.Secret, testServiceAccount *corev1.ServiceAccount) error {
+	ginkgo.By(fmt.Sprintf("Initializing Shoot %s/%s and its Client", *shootNamespace, *shootName))
+	if err := initShootAndClient(ctx, t); err != nil {
+		return err
+	}
+	t.SeedShootNamespace = ComputeTechnicalID(t.GardenerFramework.ProjectNamespace, &t.Shoot)
+
+	ginkgo.By(fmt.Sprintf("Initializing source Seed %s, target Seed %s, and their Clients", *t.Shoot.Spec.SeedName, *targetSeedName))
+	if err := initSeedsAndClients(ctx, t); err != nil {
+		return err
+	}
+
+	ginkgo.By("Fetching the objects that will be used for comparison")
+	if err := t.PopulateBeforeMigrationComparisonElements(ctx); err != nil {
+		return err
+	}
+
+	ginkgo.By("Creating test Secret and Service Account")
+	if err := t.ShootClient.DirectClient().Create(ctx, testSecret); err != nil {
+		return err
+	}
+	t.GardenerFramework.Logger.Infof("Secret resource %s/%s was created!", testSecret.Name, testSecret.Namespace)
+	if err := t.ShootClient.DirectClient().Create(ctx, testServiceAccount); err != nil {
+		return err
+	}
+	t.GardenerFramework.Logger.Infof("ServiceAccount resource %s/%s was created!", testServiceAccount.Name, testServiceAccount.Namespace)
+
+	ginkgo.By("Deploying Guest Book Application")
+	initializedApp, err := initGuestBookTest(ctx, t)
+	if err != nil {
+		return err
+	}
+	*guestBookApp = *initializedApp
+	guestBookApp.DeployGuestBookApp(ctx)
+	guestBookApp.Test(ctx)
+
+	return nil
+}
+
+func afterMigration(ctx context.Context, t *ShootMigrationTest, guestBookApp applications.GuestBookTest, testSecret *corev1.Secret, testServiceAccount *corev1.ServiceAccount) error {
+	if ginkgo.CurrentGinkgoTestDescription().Failed {
+		t.GardenerFramework.DumpState(ctx)
+	}
+
+	ginkgo.By("Fetching the objects that will be used for comparison...")
+	if err := t.PopulateAfterMigrationComparisonElements(ctx); err != nil {
+		return err
+	}
+
+	ginkgo.By("Checking if the test Secret and Service Account are migrated and cleaning them up...")
+	if err := t.ShootClient.DirectClient().Delete(ctx, testSecret); err != nil {
+		return err
+	}
+	t.GardenerFramework.Logger.Infof("Secret resource %s/%s was deleted!", testSecret.Name, testSecret.Namespace)
```

Comment:
I think the convention for NamespacedName is to have the namespace in the first place. See https://github.com/gardener/gardener/blob/v1.12.3/vendor/k8s.io/apimachinery/pkg/types/namespacedname.go or for example https://github.com/gardener/gardener/blob/v1.12.3/extensions/pkg/controller/infrastructure/reconciler.go#L194.

Can you adapt all such occurrences to `<namespace>/<name>`?

```suggestion
	t.GardenerFramework.Logger.Infof("Secret resource %s/%s was deleted!", testSecret.Namespace, testSecret.Name)
```